lib/travis/build/script/r.rb

```diff
@@ -164,6 +165,10 @@ def configure
               options_repos = "options(repos = c(#{repos_str}))"
               sh.cmd %Q{echo '#{options_repos}' > ~/.Rprofile.site}
 
+              # Set -Wall -pedantic in the site Makevars to catch more issues,
+              # as CRAN runs with -pedantic
+              sh.cmd %Q{echo 'PKG_CXXFLAGS=-Wall -pedantic\nPKG_CFLAGS=-Wall -pedantic' > ~/.Makevars.site}
```

Comment:
From [R Installation and Administration](https://cran.r-project.org/doc/manuals/r-release/R-admin.html#Customizing-package-compilation)
> There is also provision for a site-wide Makevars.site file under R_HOME/etc (in a sub-architecture-specific directory if appropriate). This is read immediately after Makeconf, and the path to an alternative file can be specified by environment variable R_MAKEVARS_SITE.

So the site wide Makevars is processed before the user specified Makevars.

Those other options sound like useful suggestions, but I don't think they are appropriate to enable for all packages, setting too many options will just make it harder for users to replicate travis results on their machines.